src/functions/example.ts

```diff
@@ -0,0 +1,13 @@
+export function example(a: any, b: any, c: any, d: any) {
```

Comment:
So this might look weird. When there is an `x.example` field, that is not inside `x.schema` field (so not `x.schema.example`), then:
- define `type` as one of `x.schema.type` or `x.type`
- verify if `x.example` is of `type`
This is only an example and I'm not sure if we even want to validate `x.example` field. Do we? If we do, is this the way to go? Or should it be validated somehow else?

Reply:
This is very confusing, good job for battling through. 

We want to support:

``` yaml
      parameters:
        - in: path
          name: petId
          required: true
          description: The id of the pet to retrieve
          schema:
            type: integer
          example: 3
```

``` yaml
      parameters:
        - in: path
          name: petId
          required: true
          description: The id of the pet to retrieve
          schema:
            type: integer
            example: 3
```

It's a quirk in OpenAPI that these are both allowed. I don't know why they are. But they are. :(

We should use exactly the same logic to validate either, but the function parameters are a little different. 

Check out the valid-example rule:

```
        "function": "schemaPath",
        "functionOptions": {
          "field": "example",
          "schemaPath": "$"
        }
```

The schemaPath is saying "pass in this current object", which would look like this:

            type: integer
            example: 3

The field keyword is asking for the field we want to validate, which is example, so now we are validating `3` against the schema `type: integer`. Huge success, everything works, yay. Until...

```
            schema: { whatever }
            required: true
            example: 3
```

The only problem is that example is being spotted (outside of a schema) so now the `schemaPath: $` has a bunch of keywords which arent a schema, and that freaks AJV out.

So the schemaPath function is fine, and we wont need an example function because that is too specific. You did a good job making valid-schema-example more accurate, but that is the only change required for that rule. The valid-openapi-example should include:

```
        "functionOptions": {
          "field": "example",
          "schemaPath": "$.schema"
        }

or something. Basically if an example is sitting next to a schema, we want to pass that example to that schema. 

Confused? Good, so am I. 😭

Reply:
Hehe, yes, a bit confused ;) thank you @philsturgeon for the detailed description! I shall dig into it some more! ;)